ios/browser/api/ads/brave_ads.mm

```diff
@@ -1739,6 +1745,25 @@ - (void)toggleDislikeAd:(NSString*)creativeInstanceId
                        /*intentional*/ base::DoNothing());
 }
 
+- (void)clearData:(void (^)())completion {
+  // Ensure the Brave Ads service is stopped before clearing data.
+  if ([self isServiceRunning]) {
+    return completion();
+  }
+
+  base::FilePath storage_path(base::SysNSStringToUTF8(self.storagePath));
+  base::ThreadPool::PostTaskAndReply(
+      FROM_HERE, {base::MayBlock()},
+      base::BindOnce(
+          [](base::FilePath storage_path) {
+            base::DeleteFile(storage_path.Append(kAdsDatabaseFilename));
+            base::DeleteFile(
+                storage_path.Append(brave_ads::kClientJsonFilename));
```

Comment:
We should also delete confirmations.json on iOS because a user can toggle Rewards off (they cannot reset Rewards).

Reply:
Why are we not clearing prefs?

Reply:
Fixed